payments-core/src/main/java/com/stripe/android/paymentsheet/injection/FlowControllerModule.kt

```diff
@@ -81,18 +84,26 @@ internal class FlowControllerModule {
 
     @Provides
     @Singleton
-    fun provideEventReporter(
+    fun provideAnalyticsRequestFactory(
         appContext: Context,
         lazyPaymentConfiguration: Lazy<PaymentConfiguration>
+    ) = AnalyticsRequestFactory(
+        appContext,
+        { lazyPaymentConfiguration.get().publishableKey },
+        setOf(PaymentSheetEvent.PRODUCT_USAGE)
+    )
+
+    @Provides
+    @Singleton
+    fun provideEventReporter(
+        appContext: Context,
+        analyticsRequestFactory: AnalyticsRequestFactory
     ): EventReporter {
         return DefaultEventReporter(
             mode = EventReporter.Mode.Custom,
             DefaultDeviceIdRepository(appContext, Dispatchers.IO),
             DefaultAnalyticsRequestExecutor(),
```

Comment:
`DefaultAnalyticsRequestExecutor` is provided elsewhere, it can be declared in the parameter